src/runtime/actor/handle.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use thiserror::Error;
+use tokio::sync::mpsc::UnboundedSender;
+
+use super::{
+    envelope::{DynEvent, Envelope},
+    Actor,
+};
+use crate::runtime::{registry::ScopeId, scope::ScopeView};
+
+/// Error sending a message to an actor
+#[derive(Error, Debug)]
+#[error("Error sending message to actor: {0}")]
+pub struct SendError(String);
+
+#[allow(missing_docs)]
+impl SendError {
+    pub fn new<S: Into<String>>(msg: S) -> Self {
+        Self(msg.into())
+    }
+}
+
+impl<S: Into<String>> From<S> for SendError {
+    fn from(msg: S) -> Self {
+        Self::new(msg)
+    }
+}
+
+/// An actor handle, used to send events
+#[derive(Debug)]
+pub struct Addr<A: Actor> {
```

Comment:
If this doc already uses the term `handle`, why not call this type `Handle` or even `ActorHandle`?

Reply:
I prefered the name `Addr`, this is probably the reason for the inconsistency. I think we should just rename this file to `addr.rs`, because I also always confuse it with the implementation of `HandleEvent`.